### PR TITLE
refactor(core,schemas): add user logto_data column to store mfa

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add a new column `user.logto_data` to store extra internal data for the user. This field will not be exposed to API response.

Currently, using it to store the mark of skip binding new MFA.

There is no alteraction script for existing user MFA data in `customData`, because this feature is not released yet, no need to alter.

## Testing

Local test and pass existing unit and integration tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/schemas/tables/users.sql [hunk 1]
```diff
@@ -16,6 +16,7 @@ create table users (
   application_id varchar(21),
   identities jsonb /* @use Identities */ not null default '{}'::jsonb,
   custom_data jsonb /* @use JsonObject */ not null default '{}'::jsonb,
+  logto_data jsonb /* @use JsonObject */ not null default '{}'::jsonb,
```
Comment (simeng-li) [inline packages/schemas/tables/users.sql:19]: nit: logto_config?

File: packages/schemas/alterations/next-1698728945-user-logto-data.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      alter table users
+        add column if not exists logto_data jsonb not null default '{}'::jsonb;
+    `);
```
Comment (simeng-li) [inline packages/schemas/alterations/next-1698728945-user-logto-data.ts:10]: should we also migrate the existing mfa skip data in the custom_data field?

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: packages/schemas/alterations/next-1698728945-user-logto-data.ts (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/alterations/next-1698728945-user-logto-data.ts]: this feature is not released yet, so no need to alter.

**Pull request merged**